### PR TITLE
Feature/implement traefik documentation

### PR DESCRIPTION
**Pull request by Darkweak**

### CODE REVIEW

File: deployment/traefik.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Implement Traefik into API Platform dockerized
+
+## Basic implementation
+
+[Traefik](https://traefik.io) is a reverse proxy / load balancer that's easy, dynamic, automatic, fast, full-featured, open source, production proven, providing metrics, and integrating with every major cluster technologie
```
Comment (antograssiot) [inline deployment/traefik.md]: technology

File: deployment/traefik.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Implement Traefik into API Platform dockerized
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
# Implementing Traefik Into API Platform Dockerized
```

File: deployment/traefik.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## Basic implementation
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
## Basic Implementation
```

File: deployment/traefik.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+[Traefik](https://traefik.io) is a reverse proxy / load balancer that's easy, dynamic, automatic, fast, full-featured, open source, production proven, providing metrics, and integrating with every major cluster technology
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
[Traefik](https://traefik.io) is a reverse proxy / load balancer that's easy, dynamic, automatic, fast, full-featured, open source, production proven, providing metrics, and integrating with every major cluster technology.
```

File: deployment/traefik.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+This tool will help you to define your own routes for your client, api and more generally for your containers.
+
+Use this custom API Platform docker-compose.yml file which implements ready-to-use Traefik container configuration.
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
Use this custom API Platform `docker-compose.yml` file which implements ready-to-use Traefik container configuration.  
```

File: deployment/traefik.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+[Traefik](https://traefik.io) is a reverse proxy / load balancer that's easy, dynamic, automatic, fast, full-featured, open source, production proven, providing metrics, and integrating with every major cluster technology
+
+This tool will help you to define your own routes for your client, api and more generally for your containers.
+
+Use this custom API Platform docker-compose.yml file which implements ready-to-use Traefik container configuration.  
+Override ports and add labels to tell Traefik to listen the routes mentionned and redirect routes to specified container.
+
+
+```--api``` Tell Traefik to generate a browser view to watch containers and IP/DNS associated easier
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
`--api` Tell Traefik to generate a browser view to watch containers and IP/DNS associated easier  
```

File: deployment/traefik.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+[Traefik](https://traefik.io) is a reverse proxy / load balancer that's easy, dynamic, automatic, fast, full-featured, open source, production proven, providing metrics, and integrating with every major cluster technology
+
+This tool will help you to define your own routes for your client, api and more generally for your containers.
+
+Use this custom API Platform docker-compose.yml file which implements ready-to-use Traefik container configuration.  
+Override ports and add labels to tell Traefik to listen the routes mentionned and redirect routes to specified container.
+
+
+```--api``` Tell Traefik to generate a browser view to watch containers and IP/DNS associated easier  
+```--docker``` Tell Traefik to listen docker api
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
`--docker` Tell Traefik to listen docker api  
```

File: deployment/traefik.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+This tool will help you to define your own routes for your client, api and more generally for your containers.
+
+Use this custom API Platform docker-compose.yml file which implements ready-to-use Traefik container configuration.  
+Override ports and add labels to tell Traefik to listen the routes mentionned and redirect routes to specified container.
+
+
+```--api``` Tell Traefik to generate a browser view to watch containers and IP/DNS associated easier  
+```--docker``` Tell Traefik to listen docker api  
+```--docker.domain=localhost``` The main DNS will be on localhost
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
`--docker.domain=localhost` The main DNS will be on localhost  
```

File: deployment/traefik.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+This tool will help you to define your own routes for your client, api and more generally for your containers.
+
+Use this custom API Platform docker-compose.yml file which implements ready-to-use Traefik container configuration.  
+Override ports and add labels to tell Traefik to listen the routes mentionned and redirect routes to specified container.
+
+
+```--api``` Tell Traefik to generate a browser view to watch containers and IP/DNS associated easier  
+```--docker``` Tell Traefik to listen docker api  
+```--docker.domain=localhost``` The main DNS will be on localhost  
+```labels:``` Key for Traefik configuration into Docker integration
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
`labels:` Key for Traefik configuration into Docker integration  
```

File: deployment/traefik.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+Use this custom API Platform docker-compose.yml file which implements ready-to-use Traefik container configuration.  
+Override ports and add labels to tell Traefik to listen the routes mentionned and redirect routes to specified container.
+
+
+```--api``` Tell Traefik to generate a browser view to watch containers and IP/DNS associated easier  
+```--docker``` Tell Traefik to listen docker api  
+```--docker.domain=localhost``` The main DNS will be on localhost  
+```labels:``` Key for Traefik configuration into Docker integration  
+```
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
```yaml
```

File: deployment/traefik.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+services:
+  ...
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
  # ...
```

File: deployment/traefik.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+  api:
+    labels: 
+      - "traefik.frontend.rule=Host:api.localhost"
+``` 
+The api DNS will be specified with traefik.frontend.rule=Host:your.host (here api.localhost)
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
The api DNS will be specified with `traefik.frontend.rule=Host:your.host` (here `api.localhost`)  
```

File: deployment/traefik.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+```
+services:
+  ...
+  api:
+    labels: 
+      - "traefik.frontend.rule=Host:api.localhost"
+``` 
+The api DNS will be specified with traefik.frontend.rule=Host:your.host (here api.localhost)  
+
+```--traefik.port=3000``` Port specified to Traefik will be exopsed by container (here React app expose the 3000 port)
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
`--traefik.port=3000` Port specified to Traefik will be exopsed by container (here React app expose the 3000 port)  
```

File: deployment/traefik.md [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+    build:
+      context: ./api
+    depends_on:
+      - db
+    env_file:
+      - ./api/.env
+     Comment out these volumes in production
+    volumes:
+      - ./api:/srv/api:rw,cached
+       If you develop on Linux, uncomment the following line to use a bind-mounted host directory instead
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
      # If you develop on Linux, uncomment the following line to use a bind-mounted host directory instead
```

File: deployment/traefik.md [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+    build:
+      context: ./api
+    depends_on:
+      - php
+     Comment out this volume in production
+    volumes:
+      - ./api/public:/srv/api/public:ro
+
+  db:
+     In production, you may want to use a managed database service
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
     # In production, you may want to use a managed database service
```

Comment (dunglas) [reply deployment/traefik.md]: Same for all other comments of this file

File: deployment/traefik.md [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+      - "traefik.port=3000"
+      - "traefik.frontend.rule=Host:localhost"
+
+volumes:
+  db-data: {}
+```
+
+Don't forget the db-data, then database won't work in this dockerized solution.
+
+```localhost``` is a reserved domain referred in your ```/etc/hosts```.
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
`localhost` is a reserved domain referred in your `/etc/hosts`. 
```

File: deployment/traefik.md [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+      - "traefik.frontend.rule=Host:localhost"
+
+volumes:
+  db-data: {}
+```
+
+Don't forget the db-data, then database won't work in this dockerized solution.
+
+```localhost``` is a reserved domain referred in your ```/etc/hosts```. 
+If you want to implement custom DNS such as production DNS in local, just put them at the end of your ```/etc/host``` file like that :
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
If you want to implement custom DNS such as production DNS in local, just put them at the end of your `/etc/hosts` file like that: 
```

File: deployment/traefik.md [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+Don't forget the db-data, then database won't work in this dockerized solution.
+
+```localhost``` is a reserved domain referred in your ```/etc/hosts```. 
+If you want to implement custom DNS such as production DNS in local, just put them at the end of your ```/etc/host``` file like that : 
+
+```
+# /etc/hosts
+...
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
# ...
```

File: deployment/traefik.md [hunk 19]
```diff
@@ -0,0 +1,122 @@
...
+
+127.0.0.1       your.domain.com
+```
+
+If you do that, you'll have to update the nelmio part in your```api/.env``` and accept the URL specified
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
If you do that, you'll have to update the `CORS_ALLOW_ORIGIN` environment variable `api/.env` to accept the specified URL.
```

File: deployment/traefik.md [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+
+## Known problems
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
## Known Issues
```

File: deployment/traefik.md [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+
+Take care of your network, if it's a type B network it may not work because containers are in type B network and it will be in conflict with Traefik sub-network
```
Comment (dunglas) [inline deployment/traefik.md]:
```suggestion
If your network is of type B, it may conflict with the traefik sub-network.
```

**Comment by Darkweak** [comment]
ping @dunglas

**Comment by dunglas** [comment]
Thanks @Darkweak!

**Pull request merged**